### PR TITLE
fix: do not use the Array.prototype.flat function

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

In some NodeJS version we cannot use `Array.prototype.flat` function. I fix that and switch to implementation with `push` function.

**Related issue(s)**
Example problem in this PR https://github.com/asyncapi/website/pull/579

### CODE REVIEW

File: lib/asyncapiSchemaFormatParser.js [hunk 1]
```diff
@@ -43,15 +43,20 @@ async function parse({ message, originalAsyncAPIDocument, fileFormat, parsedAsyn
...
+      `application/vnd.aai.asyncapi;version=${version}`,
+      `application/vnd.aai.asyncapi+json;version=${version}`,
+      `application/vnd.aai.asyncapi+yaml;version=${version}`,
+    ]);
+  });
+  mimeTypes.push(...[
+    'application/schema;version=draft-07',
+    'application/schema+json;version=draft-07',
+    'application/schema+yaml;version=draft-07',
+  ]);
```
Comment (jonaslagoni) [inline lib/asyncapiSchemaFormatParser.js:58]: You could simplify it a bit and do this: 

```suggestion
  const mimeTypes = ['application/schema;version=draft-07',
    'application/schema+json;version=draft-07',
    'application/schema+yaml;version=draft-07'];

  ['2.0.0', '2.1.0', '2.2.0', '2.3.0'].forEach(version => {
    mimeTypes.push(
      `application/vnd.aai.asyncapi;version=${version}`,
      `application/vnd.aai.asyncapi+json;version=${version}`,
      `application/vnd.aai.asyncapi+yaml;version=${version}`
    );
  });
```

Comment (magicmatatjahu) [reply lib/asyncapiSchemaFormatParser.js:58]: I wanna have `application/schema;version=draft-07` etc at the end of array as in original array :)

Comment (magicmatatjahu) [reply lib/asyncapiSchemaFormatParser.js:58]: but ok :)

Comment (jonaslagoni) [reply lib/asyncapiSchemaFormatParser.js]: Ahh right 🙂 So this? 

```suggestion
  const mimeTypes = [];

  ['2.0.0', '2.1.0', '2.2.0', '2.3.0'].forEach(version => {
    mimeTypes.push(
      `application/vnd.aai.asyncapi;version=${version}`,
      `application/vnd.aai.asyncapi+json;version=${version}`,
      `application/vnd.aai.asyncapi+yaml;version=${version}`
    );
  });
  mimeTypes.push('application/schema;version=draft-07',
    'application/schema+json;version=draft-07',
    'application/schema+yaml;version=draft-07'
  );
```

Comment (magicmatatjahu) [reply lib/asyncapiSchemaFormatParser.js]: Ok, we can have that `application/schema;version=draft-07` at the beginning :)

Comment (magicmatatjahu) [reply lib/asyncapiSchemaFormatParser.js]: I propagated suggestions.

**Comment by jonaslagoni** [comment]
I guess we cant test this can we? To ensure this cannot be introduced again?

**Comment by magicmatatjahu** [comment]
> I guess we cant test this can we? To ensure this cannot be introduced again?

In the current implementation (with `flat`) it pass in all tests, so 🤷🏼  Probably we should downgrade NodeJS version to test it, but is it worth?

**Comment by jonaslagoni** [comment]
> In the current implementation (with `flat`) it pass in all tests, so 🤷🏼 Probably we should downgrade NodeJS version to test it, but is it worth?

Depends on why we think the website deployment failed and whether it's a use-case we have to account for in the parser. Maybe create a follow-up issue afterward?

**Review by jonaslagoni — APPROVED** [review]

**Comment by magicmatatjahu** [comment]
> Depends on why we think the website deployment failed and whether it's a use-case we have to account for in the parser. Maybe create a follow-up issue afterward?

Maybe we should run workflows as matrix, but not for systems but for NodeJS versions, in this repo and in others? cc 
@derberg

**Pull request merged**